controllers/provider-aws/pkg/controlplane/actuator.go

```diff
@@ -0,0 +1,359 @@
+					SigningCA:    cas[gardencorev1alpha1.SecretNameCACluster],
+				},
+				KubeConfigRequest: &secrets.KubeConfigRequest{
+					ClusterName:  clusterName,
+					APIServerURL: common.KubeAPIServerDeploymentName,
+				},
+			},
+			&secrets.ControlPlaneSecretConfig{
+				CertificateSecretConfig: &secrets.CertificateSecretConfig{
+					Name:       common.CloudControllerManagerServerName,
+					CommonName: common.CloudControllerManagerDeploymentName,
+					DNSNames:   util.DNSNamesForService(common.CloudControllerManagerDeploymentName, clusterName),
+					CertType:   secrets.ServerCert,
+					SigningCA:  cas[gardencorev1alpha1.SecretNameCACluster],
+				},
+			},
+		}
+	},
+}
+
+var configChart = &chart.Chart{
+	Name: "cloud-provider-config",
+	Path: filepath.Join(aws.ChartsPath, "cloud-provider-config"),
+	Objects: []*chart.Object{
+		{
+			Type: &corev1.ConfigMap{},
+			Name: common.CloudProviderConfigName,
+		},
+	},
+}
+
+const (
+	// AccessKeyID is a constant for the key in a cloud provider secret and backup secret that holds the AWS access key id.
+	AccessKeyID = "accessKeyID"
+	// SecretAccessKey is a constant for the key in a cloud provider secret and backup secret that holds the AWS secret access key.
+	SecretAccessKey = "secretAccessKey"
+)
+
+var ccmChart = &chart.Chart{
+	Name:   "cloud-controller-manager",
+	Path:   filepath.Join(aws.ChartsPath, "cloud-controller-manager"),
+	Images: []string{common.HyperkubeImageName},
+	ValuesFunc: func(clusterName string, shoot *gardenv1beta1.Shoot, checksums map[string]string) (map[string]interface{}, error) {
+		s := &extensionscontroller.Shoot{Shoot: shoot}
+		return map[string]interface{}{
+			"cloudProvider":     "aws",
+			"clusterName":       clusterName,
+			"kubernetesVersion": s.GetKubernetesVersion(),
+			"podNetwork":        s.GetPodNetwork(),
+			"replicas":          s.GetReplicas(1),
+			"podAnnotations": map[string]interface{}{
+				"checksum/secret-cloud-controller-manager":        checksums[common.CloudControllerManagerDeploymentName],
+				"checksum/secret-cloud-controller-manager-server": checksums[common.CloudControllerManagerServerName],
+				"checksum/secret-cloudprovider":                   checksums[common.CloudProviderSecretName],
+				"checksum/configmap-cloud-provider-config":        checksums[common.CloudProviderConfigName],
+			},
+			"configureRoutes": false,
+			"environment": []map[string]interface{}{
+				{
+					"name": "AWS_ACCESS_KEY_ID",
+					"valueFrom": map[string]interface{}{
+						"secretKeyRef": map[string]interface{}{
+							"key":  AccessKeyID,
+							"name": common.CloudProviderSecretName,
+						},
+					},
+				},
+				{
+					"name": "AWS_SECRET_ACCESS_KEY",
+					"valueFrom": map[string]interface{}{
+						"secretKeyRef": map[string]interface{}{
+							"key":  SecretAccessKey,
+							"name": common.CloudProviderSecretName,
+						},
+					},
+				},
+			},
+			// TODO Use the higher resource limits below only for the "shooted seed" case
```

Comment:
This TODO can be removed, right?

Reply:
I thought we might actually want to properly take care of the "shooted seed" case some day. But you are right, since it's not clear how to do it this TODO is rather misplaced. Removed.